src/structurizr/model/model.py

```diff
@@ -209,38 +209,29 @@ def add_software_system(self, software_system=None, **kwargs) -> SoftwareSystem:
         self.software_systems.add(software_system)
         return software_system
 
-    def add_container(
-        self, container: Optional[Container] = None, **kwargs
-    ) -> Container:
+    def add_container(self, container: Container) -> Container:
```

Comment:
I guess there's a question now as to whether there's any point having `Model.add_container()` and `Model.add_component()` or whether it'd be better for `Container` and `Component` just to call `Model._add_element()` directly.

Reply:
I do think we should be consistent with either the `__iadd__` method calling the `add_*` methods or the other way around. (See `SoftwareSystem` below.) We could add a 'protected' helper method(s) called by both.

Reply:
Have a look at what I've done to Model in my latest commit and let me know if you think this approach is good.